### ISSUE TITLE
air gap RKE2 downstream cluster fails to pull images if the registry mirrors endpoint does not contain a schema

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#46090]
**Rancher Server Setup**
- Rancher version: v2.8.3
- Installation option (Docker install/Helm Chart):
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: v1.28.9-rke2r1  
- Cluster Type (Local/Downstream): Downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions:

**Describe the bug**

When provisioning an air-gapped RKE2 cluster, if the user sets the mirror endpoint to their private registry, but without the `https://` scheme, the cluster will fail to provision with the following error message observed in the `rancher-system-agent` logs:

```
Jun 17 13:23:56 qchcon201 rancher-system-agent[20349]: time="2024-06-17T13:23:56+02:00" level=info msg="Pulling image registry.rancher.com/rancher/system-agent-installer-rke2:v1.28.9-rke2r1"
Jun 17 13:23:56 qchcon201 rancher-system-agent[20349]: time="2024-06-17T13:23:56+02:00" level=warning msg="Ignoring relative endpoint URL for registry registry.rancher.com: \"registry123.sample.com\""
Jun 17 13:23:56 qchcon201 rancher-system-agent[20349]: time="2024-06-17T13:23:56+02:00" level=warning msg="Failed to get image from endpoint: Get \"https://registry.rancher.com/v2/\": dial TCP <redacted>:443: connect: connection refused"
```

**To Reproduce**

Create a DS RKE2 cluster and configure the registries similar to the following, 
```
    registries:
      configs: {}
      mirrors:
        registry.rancher.com:
          endpoint:
            - registry123.sample.com
```

**Result**

The cluster provision fails, and we can find the warning and error message in `rancher-system-agent` as described above. 

**Expected Result**

The cluster provision should succeed, the cluster should be able to pull images from the mirror. 

**Screenshots**

**Additional context**

- The most recent rancher-system-agent release (v0.3.6) uses wharfie v0.6.2 as the image pulling library.
- wharfie v0.6.2 still requires an absolute mirror endpoint URL including the scheme (https://github.com/rancher/wharfie/blob/v0.6.2/pkg/registries/registries.go#L167).
- The most recent release of wharfie (v0.6.6) drops this requirement, supporting endpoints without a scheme (https://github.com/rancher/wharfie/commit/506419fe098e1da1aa44e5ce5e99e608ce318c21).
- A PR to update wharfie in the system-agent remains open at https://github.com/rancher/system-agent/pull/143 

SURE-8588

**Comment by jiaqiluo** [comment]
Here is how I reproduce the bug on a normal setup Rnacher v2.8.5

Steps:
- go to `global setting`, and change `system-default-registry` to some invalid address, such as "invaid.registry.com` 
- create a DS RKE2 cluster, and configure the Registry to the following 
```
      mirrors:
        '*':
          endpoint:
            - registry.hub.docker.com
```
- create the cluster
- on the DS's node, run `journalctl -f -u rancher-system-agent` to check logs 

Result:
See the following in the logs:
```
Jul 25 19:03:06 rke2-jiaqi-a-8ddff622-bg9tk rancher-system-agent[706685]: time="2024-07-25T19:03:06Z" level=info msg="Pulling image invaid.registry.com/rancher/system-agent-installer-rke2:v1.28.11-rke2r1"
Jul 25 19:03:06 rke2-jiaqi-a-8ddff622-bg9tk rancher-system-agent[706685]: time="2024-07-25T19:03:06Z" level=warning msg="Ignoring relative endpoint URL for registry rinvaid.registry.com: \"registry.hub.docker.com\""
Jul 25 19:03:06 rke2-jiaqi-a-8ddff622-bg9tk rancher-system-agent[706685]: time="2024-07-25T19:03:06Z" level=warning msg="Failed to get image from endpoint: Get \"https://invaid.registry.com/v2/\": dial tcp: lookup rinvaid.registry.com on 127.0.0.53:53: no such host"
```

The line "Ignoring relative endpoint URL ...." is what we expect to see.

**Comment by jiaqiluo** [comment]
waiting for cutting a new RC in rancher/system-agent and bump it in rancher/rancher

**Comment by jiaqiluo** [comment]
wait for new images to be built at https://github.com/rancher/rancher/actions/runs/10206205050

**Comment by markusewalker** [comment]
**QA TEST PLAN**

|#|Scenario|
|---|---|
|1|Provision RKE2 air-gap downstream cluster with `https://` schema in mirror endpoint|
|2|Provision RKE2 air-gap downstream cluster without `https://` schema in mirror endpoint|
|3|Provision RKE2 non air-gap downstream cluster with `https://` schema in mirror endpoint|
|4|Provision RKE2 non air-gap downstream cluster without `https://` schema in mirror endpoint|

**Comment by jiaqiluo** [comment]
the fix can be validated on the lastest v2.9-head image

**Comment by markusewalker** [comment]
Validated that this is addressed on `v2.9.1-alpha1`. See details below:

**ENVIRONMENT DETAILS**
- Rancher install: Helm
- Rancher version: `v2.9.1-alpha1`

**TEST RESULT**
|#|Scenario|Result|
|---|---|---|
|1|Provision RKE2 air-gap downstream cluster with https:// schema in mirror endpoint|:white_check_mark:|
|2|Provision RKE2 air-gap downstream cluster without https:// schema in mirror endpoint|:white_check_mark:|

**VALIDATION STEPS**

**Scenario 1**
1. Provisioned downstream RKE2 `v1.30.2+rke2r1` custom cluster (1 etcd, 1 cp, 1 worker).
     - Ensured that the private registry had `https://` schema present in the endpoint.
2. Validated that the cluster and nodes came up `Active`.

_______

**Scenario 2**
1. Repeated scenario 1, but did not use `https://` schema in the registry endpoint.
